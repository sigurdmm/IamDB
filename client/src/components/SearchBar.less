@import '../constants';

.searchbar {

  background: @white;
  box-shadow: 0 6px 8px rgba(102, 119, 136, .03), 0 1px 2px rgba(102, 119, 136, .3);
  display: flex;
  height: 48px;
  width: 100%;
  align-items: center;
  cursor: text;

  &__field {
    font-family: arial,sans-serif;
    font-size: 1em;
    padding: 16px;
    height: 64px;
    background: 0 0;
    border: 0;
    flex: 1 1 auto;
    font-weight: bold;
    color: @textColor;
    min-width: 200px;
  }
  &__field:focus {
    outline: none;
  }

  &__submit {
    background: none;
    border: none;
    img{
      width: 32px;
      height: 32px;
    }
  }
}

.error {
  color: @errorRed;
  margin: 10px;
}
