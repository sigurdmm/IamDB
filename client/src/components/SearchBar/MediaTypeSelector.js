import PropTypes from 'prop-types';
import Select from 'react-select';
import React from 'react';

const options = [
  { value: 'all', label: 'All' },
  { value: 'movies', label: 'Movies' },
  { value: 'series', label: 'TV Shows' },
];

export default class MediaTypeSelector extends React.Component {
  static propTypes = {
    selectedMediaType: PropTypes.shape({
      value: PropTypes.string.isRequired,
      label: PropTypes.string.isRequired,
    }).isRequired,
    onSelectorChange: PropTypes.func.isRequired,
  };

  render() {
    return <Select
      value={this.props.selectedMediaType}
      onChange={obj => this.props.onSelectorChange(obj)}
      options={options}
    />;
  }
}
