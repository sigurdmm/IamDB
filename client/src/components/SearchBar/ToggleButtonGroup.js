import React, { PureComponent } from 'react';
import './ToggleButtonGroup.less';
import PropTypes from 'prop-types';

const ToggleButton = ({ children, onClick, active }) => <button
  onClick={onClick}
  className={`togglebuttons__button${active ? ' togglebuttons__button--toggled' : ''}`}>{children}</button>;

ToggleButton.propTypes = {
  children: PropTypes.any.isRequired,
  onClick: PropTypes.func.isRequired,
  active: PropTypes.bool.isRequired,
};

export default class ToggleButtonGroup extends PureComponent {
  static propTypes = {
    toggled: PropTypes.number.isRequired,
    buttons: PropTypes.arrayOf(PropTypes.shape({
      label: PropTypes.string.isRequired,
      value: PropTypes.string.isRequired,
    })),
    onToggle: PropTypes.func.isRequired,
  };

  onChange = id => () => {
    if (id === this.props.toggled) {
      return;
    }
    this.props.onToggle(id);
  };

  render() {
    return <div className="togglebuttons">
      {this.props.buttons.map((button, i) => <ToggleButton
        key={`togglebutton-${i}`}
        active={this.props.toggled === i}
        onClick={this.onChange(i)}>{button.label}</ToggleButton>)}
    </div>;
  }
}
