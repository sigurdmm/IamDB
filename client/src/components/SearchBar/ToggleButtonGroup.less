@import '../../constants';

.togglebuttons__container {
  display: block;

  position: relative;
}

.togglebuttons__container :first-child {
  border-radius: 2px 0 0 2px;
  border-right: none;
}

.togglebuttons__container :last-child {
  border-radius: 0 2px 2px 0;
}

.toggleButton {
  font-size: 1rem;
  line-height: 1.375rem;
  font-weight: 400;
  justify-content: center;
  align-items: center;
  background-color: @white;
  border: 1px solid @borderColor;
  cursor: pointer;
  min-width: 6.25rem;
  height: 2rem;
  padding-left: 1rem;
  padding-right: 1rem;
}

button {
  font-family: sans-serif;
  font-size: 100%;
  line-height: 1.15;
  margin: 0;
  overflow: visible;
}

.toggleButtonGroup__toggled {
  background-color: @primaryColor;
  color: @white;
  border-color: @primaryColor;
}
