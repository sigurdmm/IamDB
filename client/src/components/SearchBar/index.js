import React from 'react';
import {
  Formik,
  Form,
  Field,
  ErrorMessage,
} from 'formik';
import './index.less';
import PropTypes from 'prop-types';
import icon from './search.svg';

const SearchBar = ({ onSubmit }) => <div>
    <Formik
      initialValues={{ search: '' }}
      validate={(values) => {
        const errors = {};
        if (!values.search) {
          errors.search = 'Search input can\'t be empty';
        }
        return errors;
      }}
      onSubmit={(values) => {
        onSubmit(values.search);
      }}
    >
      <div>
        <Form className="searchbar">
          <Field
            type="field"
            className="searchbar__field"
            placeholder="The dark knight rises"
            name="search"
            autoFocus/>
          <button type="submit" className="searchbar__submit">
            <img src={icon}/>
          </button>
        </Form>
        <ErrorMessage className={'error'} name="search" component="div"/>
      </div>
    </Formik>
  </div>;

SearchBar.propTypes = {
  onSubmit: PropTypes.func.isRequired,
};

export default SearchBar;
