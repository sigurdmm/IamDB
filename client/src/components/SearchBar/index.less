@import '../../constants';

.searchbar {
  .shadow-1;
  background: @white;
  border-radius: 2px;
  display: flex;
  width: 100%;
  margin: 1em 0;
  align-items: center;
  cursor: text;

  position: relative;

  &__field {
    font-size: 1em;
    box-sizing: border-box;
    padding: 0.5em;
    height: 64px;
    background: 0 0;
    border: 0;
    flex: 1 1 auto;
    font-weight: bold;
    color: @textColor;
    min-width: 200px;

    position: relative;

    &::placeholder,
    &::-ms-input-placeholder {
      color: #bbbbbb;
    }
  }
  &__field:focus {
    outline: none;
  }

  &__submit {
    box-sizing: border-box;
    background: transparent;
    border: none;

    cursor: pointer;

    padding: 0.75em 1em;
    position: relative;

    &:hover {
      background-color: lighten(@primaryColor, 78%);
    }

    img {
      width: 20px;
      height: 20px;
    }
  }
}

.error {
  color: @errorRed;
  margin: 10px;
}
