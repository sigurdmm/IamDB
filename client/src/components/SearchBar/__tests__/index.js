import React from 'react';
import { shallow } from 'enzyme';

import SearchBar from '../index';

describe('<SearchBar/>', () => {
  it('should load properly', () => {
    const wrapper = shallow(<SearchBar onSubmit={jest.fn()}/>);

    expect(wrapper).toMatchSnapshot();
  });

  // it('should call onSubmit on submit', () => {
  //   const wrapper = mount(<SearchBar onSubmit={jest.fn()}/>);
  // const input = wrapper.find('input');
  // input.simulate('change', { target: { value: 'Hello' } });
  // console.log(wrapper.find('input').value);
  // wrapper.find('.searchbar__field').at(1).props().formik.)
  // console.log(wrapper.find('.searchbar__field').at(1).props().formik);
  // const input = wrapper.find('.searchbar__field').at(1);
  // const form = wrapper.querySelector('searchbar')
  // console.log(input);
  // input.simulate('change', { target: { name: 'search', values: { search: 'abc' } } });
  // console.log(input.props());
  //
  // expect(wrapper).toMatchSnapshot();
//   });
});


// describe('MyForm', () => {
//   test('should update an input when it is changed', () => {
//     const spy = jest.fn();
//     const tree = shallow(<SearchBar onSubmit={spy} />);
//
//     const form = tree.find('.searchbar__field');
//
//     console.log('form');
//     console.log(form);
//     // console.log(form.instance());
//
//     expect(false).toBe(tree.find('.searchbar__submit').find('button').simulate('press'));
//
//     form.simulate('change', {
//       // you must add this next line as (Formik calls e.persist() internally)
//       persist: () => {},
//       // simulate changing e.target.name and e.target.value
//       target: {
//         name: 'search',
//         value: 'ian',
//       },
//     });
//
//     tree.find('.searchbar').simulate('onSubmit', {
//       preventDefault: () => {},
//     });
//
//     expect(false).toBe(spy.mock.calls);
//   });
// });
