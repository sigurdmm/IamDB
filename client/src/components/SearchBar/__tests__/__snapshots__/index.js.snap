// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchBar/> should load properly 1`] = `
<div
  className="filterandsearch"
>
  <ToggleButtonGroup
    buttons={
      Array [
        Object {
          "label": "1",
          "value": "2",
        },
        Object {
          "label": "3",
          "value": "4",
        },
      ]
    }
    onToggle={[MockFunction]}
    toggled="2"
  />
  <Formik
    enableReinitialize={false}
    initialValues={
      Object {
        "search": "",
      }
    }
    isInitialValid={false}
    onSubmit={[Function]}
    validate={[Function]}
    validateOnBlur={true}
    validateOnChange={true}
  >
    <div>
      <Form
        className="searchbar"
      >
        <C
          autoFocus={true}
          className="searchbar__field"
          name="search"
          placeholder="The dark knight rises"
          type="field"
        />
        <button
          className="searchbar__submit"
          type="submit"
        >
          <img
            src="search.svg"
          />
        </button>
      </Form>
      <C
        className="error"
        component="div"
        name="search"
      />
    </div>
  </Formik>
</div>
`;
